ImageData/Hdf5Loader.h

```diff
@@ -43,6 +47,9 @@ class Hdf5Loader : public FileLoader {
     casacore::ArrayBase* GetStatsData(FileInfo::Data ds) override;
 
     casacore::Lattice<float>* LoadSwizzledData(FileInfo::Data ds);
+    // current region states
+    std::unordered_map<int, RegionState> _region_states;
```

Comment:
I would prefer it if you passed in references to the original properties on the frame object instead of duplicating them here (and changing the state in two different places from the frame). The tidiest way to do this may be to pass in a reference to the entire frame object, and calling a (new) public getter on the frame to do this check. You could call the same getter from the non-swizzled calculation. Rob has some concerns that values may be cached if the compiler uses certain optimisations, so you may want to test this with different build types and possibly use the `volatile` keyword for the properties.

Reply:
Now I move the region and cursor states variables from the Frame to FileLoader. The states check functions are also moved to the FileLoader. So they can be called from the Frame or the FileLoader itself. The states variables are set as  `volatile` now, including `RegionState`, `RegionConfig`, `CursorXy`, and `_connected`.

Reply:
It doesn't make any sense for these variables to be on the loader. They have nothing to do with loading images. I strongly suggest that you leave them on the frame object, where they were before, and pass some kind of reference into the loader calculation function so that they can be queried from that function. I still believe that the cleanest way to do this is to add a public getter to the frame object and to pass in a reference to the frame object to the calculation function so that that getter can be called.